Classes/Utility/LoginFailureUtility.php

```diff
@@ -48,30 +29,21 @@ public static function failuresEqual($amount)
         $amount = (int) $amount;
 
         if (!isset(static::$register[$amount])) {
-            $table = 'sys_log';
+            $ip = \TYPO3\CMS\Core\Utility\GeneralUtility::getIndpEnv('REMOTE_ADDR');
 
-            $queryBuilder = self::getQueryBuilderForTable($table);
-            $queryBuilder->getRestrictions()->removeAll();
-            $expression = $queryBuilder->expr();
+            $queryBuilder = self::getQueryBuilderForTable('sys_log');
             $rows = $queryBuilder
                 ->select('error')
-                ->from($table)
+                ->from('sys_log')
                 ->where(
-                    $expression->eq('type', $queryBuilder->createNamedParameter(255, \PDO::PARAM_INT)),
-                    $expression->in(
+                    $queryBuilder->expr()->eq('type', 255),
+                    $queryBuilder->expr()->in(
                         'details_nr',
-                        $queryBuilder->createNamedParameter([1,2], Connection::PARAM_INT_ARRAY)
+                        $queryBuilder->createNamedParameter([1, 2], \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
                     ),
-                    $expression->eq(
-                        'IP',
-                        $queryBuilder->createNamedParameter(
-                            \TYPO3\CMS\Core\Utility\GeneralUtility::getIndpEnv('REMOTE_ADDR'),
-                            \PDO::PARAM_STR
-                        )
-                    ),
-                    $expression->gt('tstamp', $queryBuilder->createNamedParameter(time() - 86400, \PDO::PARAM_INT))
```

Comment:
Is there any reason for ignoring the timestamp??

Reply:
The timestamp is taken care by the order. It doesnt matter if one enters 5 times in a row the wrong login or leaves a month between the attempts.